hazelcast/src/main/java/com/hazelcast/nio/tcp/TcpIpConnection.java

```diff
@@ -240,24 +240,38 @@ public void close(String reason, Throwable cause) {
             logger.warning(e);
         }
 
-        Object connAddress = getConnectionAddress();
-        String message = "Connection [" + connAddress + "] lost. Reason: ";
-        if (closeReason != null) {
-            message = closeReason;
-        } else if (cause == null) {
-            message += "Socket explicitly closed";
-        } else {
-            message += cause.getClass().getName() + "[" + cause.getMessage() + "]";
-        }
-
-        logger.info(message);
         connectionManager.onClose(this);
         connectionManager.getIoService().onDisconnect(endPoint, cause);
         if (cause != null && monitor != null) {
             monitor.onError(cause);
         }
     }
 
+    private void logClose() {
+        String message = toString() + " closed. Reason: ";
+        if (closeReason != null) {
+            message += closeReason;
+        } else if (closeCause != null) {
```

Comment:
The cause already is included in the logging by adding the stacktrace of the closeCause. 

```
  private void logClose() {
        String message = toString() + " closed. Reason: ";
        if (closeReason != null) {
            message += closeReason;
        } else if (closeCause != null) {
            message += closeCause.getClass().getName() + "[" + closeCause.getMessage() + "]";
        } else {
            message += "Socket explicitly closed";
        }

        if (connectionManager.getIoService().isActive()) {
            if (closeCause == null || closeCause instanceof EOFException) {
                logger.info(message);
            } else {
                logger.warning(message, closeCause);<-------
            }
        } else {
            if (closeCause == null) {
                logger.finest(message);
            } else {
                logger.finest(message, closeCause);
            }
        }
    }
```